### ISSUE TITLE
behind-the-scene inheritance

### ISSUE THREAD

**Issue by dezhavu** [gorhill/uMatrix#883]
Is it possible to an option for advanced users, so that behind-the-scene and other schemes (file-scheme, about-scheme, data-scheme etc) do NOT inherit from Global Scope (*), as for me I use Legacy uMatrix in unbranded build of FF 57 and it's very inconvenient to block those rules (or parse them thus increasing config file size for no reason)
A little option could be usefull in future.

Thanks for your work & time.

**Comment by dezhavu** [comment]
<a href="http://www.imagebam.com/image/3f24bb687319453" target="_blank"><img src="http://thumbs2.imagebam.com/7e/ee/6f/3f24bb687319453.jpg" alt="imagebam.com"></a>

<a href="http://www.imagebam.com/image/091777687319493" target="_blank"><img src="http://thumbs2.imagebam.com/f1/a8/1a/091777687319493.jpg" alt="imagebam.com"></a>

**Comment by gorhill** [comment]
More generally, you propose the ability to isolate a scope from the parent one.

**Comment by dezhavu** [comment]
As you could have noticed on the pictures attached, i use dynamic ruleset, so actually it's not a big deal to block those Global Scope domains for behind-the-scene, but the idea is that behind-the-scene should't inherit from Global Scope, IF there's a rule for that scheme:

For instance:
* * * block
* vimeo.com * allow
* vimeo.com frame allow
behind-the-scene * * block

Now vimeo.com doesn't care about behind-the-scene * * block, because it's allowed in Global Scope..
I saw numerous of samples like this in the logger, when requests from the webpage somehow appear in the behind-the-scene scheme, and if it inherits from global, omitting it's own scheme rules, thats no good for no good reason.

**Comment by dezhavu** [comment]
>More generally, you propose the ability to isolate a scope from the parent one.

could be something like this if to look at it in more global, would be even more flexible, but keeping in mind of "no new feature request" didn't structure it like this, the idea is great though

**Comment by theWalkingDuck** [comment]
What about using  `matrix-off: behind-the-scene false` instead of `behind-the-scene * * block` ?

**Comment by gorhill** [comment]
> What about using `matrix-off: behind-the-scene false` instead of behind-the-scene * * block ?

He wants the ability to modify the way rules propagate on the _z_ axis (from broader scopes to narrower ones), the `matrix-off` switch is to toggle on/off matrix filtering, I don't find this a good match for modifying z-propagation. The _all_ cell however is a good choice, there currently is no such thing as `scope * * inherit`, this rule means nothing in uMatrix. So it is available for leverage -- the _inherit_ action is already used in other cells to prevent looking up a rule from a broader scope.

**Comment by uBlock-user** [comment]
Removing the `matrix-off` rule itself would block all `behind the scenes` requests unless you whitelist a cell.

**Comment by dezhavu** [comment]
>Removing the matrix-off rule itself would block all behind the scenes requests unless you whitelist a cell.
Yep, it takes some time to get the whole thing, could be a bit complicated at first.

There's another issue that came up when I tried updating from 1.1.17rc0 (that was post rc0, about 4 commits after it, as far as i remember). Legacy doesn't work in FF5602, neither in FF57. Webextension works in both.

Console on install:
```
TypeError: vAPI.messaging.send is not a function[Learn More]  popup.js:1462:9
	<anonymous> chrome://umatrix/content/js/popup.js:1462:9
	matrixSnapshotPoller< chrome://umatrix/content/js/popup.js:1441:6
	<anonymous> chrome://umatrix/content/js/popup.js:1358:29
	<anonymous> chrome://umatrix/content/js/popup.js:30:2
```

Dashboard available only by directly going to chrome://umatrix/content/dashboard.html
Logger available only by directly going to chrome://umatrix/content/logger-ui.html

Every option is clear inside, no checkboxes, no rules, no hosts files, even nothing in About (no version)
Console:
```
TypeError: vAPI.messaging.send is not a function[Learn More]  about.js:133:5
	<anonymous> chrome://umatrix/content/js/about.js:133:5
	<anonymous> chrome://umatrix/content/js/about.js:122:2

TypeError: vAPI.messaging.send is not a function[Learn More]  settings.js:154:1
	<anonymous> chrome://umatrix/content/js/settings.js:154:1
	<anonymous> chrome://umatrix/content/js/settings.js:28:2

TypeError: vAPI.messaging.addListener is not a function[Learn More]  hosts-files.js:39:1
	<anonymous> chrome://umatrix/content/js/hosts-files.js:39:1
	<anonymous> chrome://umatrix/content/js/hosts-files.js:28:2

TypeError: vAPI.messaging.send is not a function[Learn More]  cloud-ui.js:207:1
	<anonymous> chrome://umatrix/content/js/cloud-ui.js:207:1
	<anonymous> chrome://umatrix/content/js/cloud-ui.js:28:2
```

**Comment by dezhavu** [comment]
<a href="http://www.imagebam.com/image/f71625688356843" target="_blank"><img src="http://thumbs2.imagebam.com/17/8a/ef/f71625688356843.jpg" alt="imagebam.com"></a> 

btw latest uBlock 1.14.22 Legacy works in both

**Comment by gorhill** [comment]
uMatrix for Legacy Firefox is no longer maintained: see #824.

**Comment by dezhavu** [comment]
Yeah, well 1.1.17rc0 works at least.
Anyway, the subj remains in Logger as requests from webpage, so if you find time pls take a look at it.

Thanks for your attention.
